### DISCUSSION TITLE
how to tell if data from cache globally? I want to collect the cache rate metrics.

### DISCUSSION THREAD

**Discussion by lili21** [TanStack/query#8404] [Q&A]
Here is my use case. I'm trying to collect some metrics , and measure the performance improvement from RQ.

Beside, I'm using the persister with RQ. and It would be helpful If I can collect the cache rate metrics and adjust the cache strategy.

**Comment by lili21** [comment]
I've tried get the info from `queryCache.subsribe` method

```js
const queryCache = new QueryCache();

queryCache.subscribe(event => {
  if (event.query.isStale()) {
  } else {
  }
})
```

but the callback function will be called multiple time for every query.

**Comment by TkDodo** [comment]
> I want to collect the cache rate metrics.

Every re-render will read data from the cache, and thus be a “cache hit”. That’s because data lives exclusively _in the cache_. I’m not really sure how you would quantify a “cache hit”? Maybe something like: “If a new component mounts and that component needs some data, but its already in the cache and not stale” ? But you have to be more specific. “Cache rate metrics” sounds like it’s a something you have to present to meet some managerial requirements but on its own, it doesn’t make much sense in this context...

Note: if what I said is what you want, filter your events down to the type `’observerAdded’`.

**Reply by lili21** [reply]
Got it. I'm more care about the page visit performance, like LCP.  the LCP should be fast if the data is from cache.  it would be nice if  I can tell it from re-render.
